src/Core/Grid/DataProvider/LogGridDataProvider.php

```diff
@@ -59,8 +60,17 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function getRows(array $filters)
+    public function getRows(SearchParametersInterface $searchParameters)
     {
+        // make search parameters compatible with logRepositoryQuery
```

Comment:
each page can have specific default values, this is why I've created a LogFilters class in my pull request on this specific subject. I can make these classes implement your interface: can you review my pr? Tomorrow we can talk about how we could improve the 2 prs to make them compatible :)

In an ideal world, we merge this one first. Then I rebase mine and I adapt it to keep the `Filters` instance part of every controller action arguments.

Then I work on the storage/delete of saved parameters and we are good for every listing page... except security but we will be iso with 1.6: it's good enough to be merged.

Reply:
Yes, this search part is really tricky at the moment. There is also duplicate for default filters. For example, each Grid definition provides default 'order by' and 'order way' but the same defaults are in ``LogsFilters``, I guess we have to remove it from grid definition?